content/tutorial-react-native/rn-03.md

```diff
@@ -0,0 +1,343 @@
+  );
+
+  _renderHeader = (trainer) => (
+    <Title trainer={this.props.data.Trainer} />
+  );
+
+  _renderSeparator = (sectionID, rowID) => (
+    <View
+      key={`${sectionID}-${rowID}`}
+      style={{ height: 1, backgroundColor: '#CCCCCC'}}
+    />
+  );
+}
+```
+
+You can then modify `Title` to accept `this.props;data.Trainer` as props `trainer` and use of 
+`this.props.trainer.ownedPokemons.length` that displays the correct amount of pokemons.
+
+To confirm your environment is all correctly setup, start the app now from XDE.
+
+
+### Dynamic Query Variables
+
+If everything working, we can now continue to implement a detailed view of a single pokemon when we click on a 
+`PokemonListItem` component. 
+
+Have a look at the `PokemonDetailScreen` directory that we prepared for you. 
+The component in `screen/PokemonDetailScreen/components/PokemonDetail.js` render the screen based on the props pokemon 
+that the page pass.
+
+We could have use the existing data from the list to rendre the whole screen, without an extra request but we will still
+do the extra request to show yout how to customize the request based on the route.
+
+We already created the navigation in `navigation/` that assigns the `PokemonDetailScreen` to the path `pokemonDetail` 
+so we can. Exponent navigation allows us to pass extra parameters to the route like this :
+
+```js
+_goToDetail = (pokemon) => {
+  this.props.navigation
+    .getNavigator('root')
+    .push(Router.getRoute('pokemonDetail', { pokemon }));
+}
+```
+
+where `getNavigator` allows to select the root navigator (and not the tab navigator) and push some routes on it, which makes
+the second order pages'tab bar disappear, and `push` allow to pass an extra object of parameters.
+
+Then we can get the paramaters like this :
+
+```js
+<PokemonDetailWithData pokemonId={this.props.route.params.pokemon.id}/>
+}
+```
+
+
+The `PokemonDetailScreen` component is responsible to pass down a pokemon to the `PokemonDetail` component. Let's add a 
+`PokemonQuery` to `PokemonDetail` that is fetching the required pokemon object:
+
+```js
+const PokemonQuery = gql`
+  query PokemonQuery($id: ID!) {
+    Pokemon(id: $id) {
+      id
+      url
+      name
+    }
+  }
+`
+```
+
+As you can see, the query requires a query variable `id` of type `ID` that we have to supply using the query option 
+`variables` as before. However, other than with the trainer name variable, we cannot just use a cannot value as the 
+id in our case. For that reason, we have the possibility to access the props when creating the `variables` object. 
+So, we can replace:
+
+```js
+export PokemonDetail;
+```
+
+with:
+
+```js
+export const PokemonDetailWithData = graphql(PokemonQuery, {
+  options: (ownProps) => {
+    return {
+      variables: {
+        id: ownProps.pokemonId
+      }
+    }
+  }
+})(PokemonDetail);
+```
+
+Note that by default Apollo try to fill unfilled `variable` with `ownProps.variable`, so this was valid too:
```

Comment:
`Note that by default Apollo tries to fill any unfilled variable called `variable` with `ownProps.variable`, so this is valid too`